lib/project_types/script/layers/domain/extension_point.rb

```diff
@@ -9,24 +9,75 @@ class ExtensionPoint
         def initialize(type, config)
           @type = type
           @deprecated = config["deprecated"] || false
-          @sdks = {
-            ts: ExtensionPointAssemblyScriptSDK.new(config["assemblyscript"]),
-          }
+          @sdks = ExtensionPointSDKs.new(config)
         end
 
         def deprecated?
           @deprecated
         end
       end
 
-      class ExtensionPointAssemblyScriptSDK
-        attr_reader :package, :version, :sdk_version, :toolchain_version
+      class ExtensionPointSDKs
```

Comment:
Just curious.  Why is this class needed?  Why not just have the `all` method defined in `ExtensionPointSDK`?  I also don't understand the value of the `assemblyscript` and `rust` methods here.  🤔

Reply:
The `ExtensionPoint#sdks` method previously was a hash, but this PR makes it an explicit class (`ExtensionPointSDKs`). It lets us add convenience methods like `all`. And instead of calling `ExtensionPoint#sdks[:assemblyscript]`, I prefer the explicit new interface of `ExtensionPoint#sdks.assemblyscript`.

`ExtensionPointSDKs` is an aggregate, whereas `ExtensionPointSDK` is basically an interface for AssemblyScript and Rust SDKs. We can't move `all` to the `ExtensionPointSDK` since it shouldn't know about its subclasses.

Reply:
I don't feel strongly about this.  But I would have been ok with having `self.all` in `ExtensionPointSDK`.  I think having a static method that knows all its subclass in this case is ok such that it functions much like a factory method.  I have seen in the past people make mistakes which would be similar to adding a new subclass of `ExtensionPointSDK`, but forgetting to update the `ExtensionPointSDKs`.  It would be good if the definition of a new subclass automatically adds to what `all` would return.  In this case, it's not so much of a problem since these classes are in the same file.